homeassistant/components/assist_pipeline/pipeline.py

```diff
@@ -1295,7 +1299,9 @@ async def process_enhance_audio(
             for dirty_chunk in chunk_samples(
                 dirty_samples, BYTES_PER_CHUNK, self.audio_chunking_buffer
             ):
-                yield self.audio_enhancer.enhance_chunk(dirty_chunk, timestamp_ms)
+                yield await self._loop.run_in_executor(
+                    None, self.audio_enhancer.enhance_chunk, dirty_chunk, timestamp_ms
+                )
```

Comment:
I didn't use `async_add_executor_job` since we should never need to wait for this job

Reply:
wouldn't calling out to the executor this often be a problem too?

Reply:
In testing it doesn't seem to call that often. I'm not sure how often it's expected to be called though so I haven't marked it as ready

Reply:
It's usually called for every 10ms of audio since that's what speex operates on. The audio often comes in as larger chunks, though, so we could be smarter about splitting it.

Reply:
Creating too many executor jobs is a concern at 10ms chunk sizes.  I'm kinda surprised it was enough to set off the loop blocking detection in asyncio debug, but I guess audio processing is cpu intensive.

Reply:
It's running [microVAD code](https://github.com/rhasspy/pymicro-vad) that I partially [hand wrote](https://github.com/rhasspy/pymicro-vad/blob/master/micro_vad/micro_array.hpp) and is probably not very well optimized.
It may be worth moving the audio processing to a separate thread where it can take in full chunks, split them into 10ms, and run them through microVAD, etc.